api/lib/authUserCAS.js

```diff
@@ -174,73 +159,46 @@ module.exports = {
       );
    },
 
-   middleware: (req, res, next, tenantUrl) => {
+   login: (req, res, tenantUrl) => {
       // Authenticate the unknown user now
 
       // Inject the AppBuilder site URL from the config into
       // the headers so that Passport CAS will know where to
       // redirect back to.
-      const siteURL = url.parse(tenantUrl);
+      const siteURL = new url.URL(tenantUrl);
       req.headers["x-forwarded-proto"] = siteURL.protocol;
       req.headers["x-forwarded-host"] = siteURL.host;
 
-      let auth = passport.authenticate("cas", (err, user, info) => {
+      const auth = passport.authenticate("cas");
+      const afterAuth = (err, ...params) => {
          // Server errors
          if (err) {
             res.serverError(err);
             authLogger(req, "CAS auth error");
             req.ab.notify.developer(err, {
                context: "CAS authentication (err)",
-               user,
-               info,
-            });
-            return;
-         }
-         if (info instanceof Error) {
-            res.serverError(info);
-            authLogger(req, "CAS auth error");
-            req.ab.notify.developer(info, {
-               context: "CAS authentication (info)",
-               user,
+               ...params,
             });
             return;
          }
          // Authentication failed
-         if (!user) {
+         if (!req.user) {
             let err = new Error("CAS Auth failed");
             res.serverError(err);
             authLogger(req, "CAS auth error");
             req.ab.notify.developer(err, {
                context: "CAS authentication failed",
-               user,
-               info,
+               ...params,
             });
             return;
          }
-
-         // CAS auth succeeded
-         req.logIn(user, (err) => {
-            // ... but the site did not?
-            if (err) {
-               res.serverError();
-               authLogger(req, "CAS auth error?");
-               req.ab.notify.developer(err, {
-                  context: "Error performing passport.logIn()",
-                  user,
-                  info,
-               });
-               return;
-            }
-
-            // Authenticated!
-            req.session.user_id = user.uuid;
-            req.session.tenant_id = req.ab.tenantID;
-            req.ab.user = user;
-         });
-
-         next();
+         // Successful auth redirect to "/"
+         res.redirect("/");
```

Comment:
Just wondering. There's currently no other page besides `/` in AppBuilder. But might we ever have a case where the user should be sent to a different page? For example, if we have a pop-up iframe reauth feature on session timeout, I'm guessing we would want the pop-up frame to load a self-closing page after login rather than `/`. 

In a multi-page app, we would redirect the user back to the page they originally requested, though that doesn't apply here.

Reply:
Yes, could be good idea for the future.